feature-libs/pickup-in-store/components/container/cart-pickup-options-container/cart-pickup-options-container.component.ts

```diff
@@ -4,44 +4,157 @@
+    protected vcr: ViewContainerRef
   ) {
     // Intentional empty constructor
   }
 
   ngOnInit() {
     this.pickupOption$ = this.outlet?.context$.pipe(
+      filter((outletContextData) => !!outletContextData),
+      tap(
+        (outletContextData) =>
+          (this.entryNumber = outletContextData.entryNumber ?? -1)
+      ),
+      tap(
+        (outletContextData) =>
+          (this.quantity = outletContextData.quantity ?? -1)
+      ),
+      tap(
+        (outletContextData) =>
+          (this.productCode = outletContextData.product?.code || '')
+      ),
+      tap(
+        (outletContextData) =>
+          (this.name = outletContextData.deliveryPointOfService?.name ?? '')
+      ),
       map(
         (item): PickupOption =>
-          item?.deliveryPointOfService ? 'pickup' : 'delivery'
+          item.deliveryPointOfService ? 'pickup' : 'delivery'
       )
     );
 
     this.displayName$ = this.outlet?.context$.pipe(
-      map((entry) => entry?.deliveryPointOfService?.name),
+      filter((outletContextData) => !!outletContextData),
+      map((entry) => entry.deliveryPointOfService?.name),
       filter((name): name is string => !!name),
-      tap((storeName) => this.storeDetails.loadStoreDetails(storeName)),
-      switchMap((storeName) => this.storeDetails.getStoreDetails(storeName)),
-      map((store) => store?.displayName ?? '')
+      tap((storeName) =>
+        this.pickupLocationsSearchService.loadStoreDetails(storeName)
+      ),
+      switchMap((storeName) =>
+        this.pickupLocationsSearchService.getStoreDetails(storeName)
+      ),
+      map((store) => store?.displayName ?? ''),
+      tap((_displayName) => _displayName && (this.displayNameIsSet = true))
+    );
+
+    this.activeCartFacade
+      .getActive()
+      .pipe(
+        tap((cart) => (this.cartId = cart.guid ?? '')),
+        tap((cart) => (this.userId = cart.user?.uid ?? ''))
+      )
+      .subscribe();
+  }
+
+  onPickupOptionChange(pickupOption: PickupOption): void {
+    const payload: SetPickupOptionDeliveryPayload = {
+      deliveryPointOfService: {
+        name: '',
+      },
+      product: {
+        code: this.productCode,
+      },
+      quantity: this.quantity as number,
+    };
+
+    this.intendedPickupLocationService.setPickupOption(
+      this.productCode,
+      pickupOption
     );
+
+    this.intendedPickupLocationService.removeIntendedLocation(this.productCode);
+
+    const preferredStore = this.preferredStoreService.getPreferredStore();
+
+    pickupOption === 'delivery' &&
+      this.pickupLocationsSearchService.setPickupOptionDelivery(
+        this.cartId,
+        this.entryNumber,
+        this.userId,
+        payload
+      );
+
+    if (!this.displayNameIsSet) {
+      this.openDialog();
+    } else if (preferredStore) {
+      this.intendedPickupLocationService.setIntendedLocation(this.productCode, {
+        ...preferredStore,
+        pickupOption: 'pickup',
+      });
+    }
+  }
```

Comment:
This method is very complex and hard to follow the logic of. It should be commented and/or split into smaller methods